mm2src/mm2_bin_lib/src/mm2_native_lib.rs

```diff
@@ -48,40 +40,75 @@ pub unsafe extern "C" fn mm2_main(conf: *const c_char, log_cb: extern "C" fn(lin
         }};
     }
 
-    if LP_MAIN_RUNNING.load(Ordering::Relaxed) {
-        eret!(MainErr::AlreadyRuns)
-    }
-    CTX.store(0, Ordering::Relaxed); // Remove the old context ID during restarts.
-
     if conf.is_null() {
-        eret!(MainErr::ConfIsNull)
+        eret!(StartupErrorCode::InvalidParams, "Configuration is null")
     }
-    let conf = CStr::from_ptr(conf);
-    let conf = match conf.to_str() {
+    let conf_cstr = CStr::from_ptr(conf);
+    let conf_str = match conf_cstr.to_str() {
         Ok(s) => s,
-        Err(e) => eret!(MainErr::ConfNotUtf8, e),
+        Err(e) => eret!(
+            StartupErrorCode::InvalidParams,
+            format!("Configuration is not valid UTF-8: {}", e)
+        ),
+    };
+
+    let conf: json::Value = match json::from_str(conf_str) {
+        Ok(v) => v,
+        Err(e) => eret!(
+            StartupErrorCode::ConfigError,
+            format!("Failed to parse configuration: {}", e)
+        ),
     };
-    let conf = conf.to_owned();
+
+    if LP_MAIN_RUNNING.load(Ordering::Relaxed) {
+        eret!(StartupErrorCode::AlreadyRunning, "MM2 is already running");
+    }
+
+    CTX.store(0, Ordering::Relaxed); // Remove the old context ID during restarts.
 
     register_callback(FfiCallback::with_ffi_function(log_cb));
+
     let rc = thread::Builder::new().name("lp_main".into()).spawn(move || {
         if let Err(true) = LP_MAIN_RUNNING.compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed) {
             log!("lp_main already started!");
             return;
         }
```

Comment:
shouldn't we propagate this as an error back to the caller? do we?

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2389/commits/f61bac7c3400186017714e6dd254cf4f00a67d43